ReactiveXComponent/RabbitMq/RabbitMqSubscriber.cs

```diff
@@ -0,0 +1,244 @@
+            RabbitMqSubscriberInfos consumerChannel;
+            _subscribers.TryGetValue(subscriberKey, out consumerChannel);
+            try
+            {
+                if (consumerChannel == null || consumerChannel.IsOpen) return;
+                if (_connection == null || !_connection.IsOpen) return;
+                var exchangeName = _xcConfiguration?.GetComponentCode(component).ToString();
+                var routingKey = _xcConfiguration?.GetSubscriberTopic(component, stateMachine);
+                var channel = _connection.CreateModel();
+                channel.ModelShutdown += ChannelOnModelShutdown;
+                channel.ExchangeDeclare(exchangeName, ExchangeType.Topic);
+                var queueName = channel.QueueDeclare().QueueName;
+                var subscriber = new EventingBasicConsumer(channel);
+                channel.BasicConsume(queueName, false, subscriber);
+                channel.QueueBind(queueName, exchangeName, routingKey, null);
+
+                var replyChannel = _connection.CreateModel();
+                replyChannel.ModelShutdown += ChannelOnModelShutdown;
+                        
+                var thread = new Thread(Listen);
+                thread.Start( new RabbitMqSubscriberInfos()
+                {
+                    Channel = channel,
+                    ReplyChannel = replyChannel,
+                    Subscriber = subscriber,
+                    IsOpen = true
+                });
+                _subscribers[subscriberKey].Channel = channel;
+                _subscribers[subscriberKey].ReplyChannel = replyChannel;
+                _subscribers[subscriberKey].IsOpen = true;
+                _subscribers[subscriberKey].Subscriber = subscriber;
+            }
+            catch (OperationInterruptedException e)
+            {
+                throw new Exception("Start consumer failure", e);
+            }
+        }
+
+        private void AddToSubscribersRepository(SubscriberKey subscriberKey)
+        {
+            if (!_subscribers.ContainsKey(subscriberKey))
+            {
+                _subscribers.Add(subscriberKey, new RabbitMqSubscriberInfos());
+            }
+        }
+
+        private void AddToCallBacksBySubscriberRepository(SubscriberKey subscriberKey, Action<MessageEventArgs> callback)
+        {
+            if (_callbacksBySubscriberKey.ContainsKey(subscriberKey))
+            {
+                _callbacksBySubscriberKey[subscriberKey].Add(callback);
+            }
+            else
+            {
+                _callbacksBySubscriberKey.Add(subscriberKey, new List<Action<MessageEventArgs>>());
+                _callbacksBySubscriberKey[subscriberKey].Add(callback);
+            }
+        }
+
+        public delegate void ConnectionFailure(object sender, StringEventArgs reason);
+
+        public event ConnectionFailure ConnectionFailed;
```

Comment:
As mentionned Nicolas, StringEventArgs is useless. You should use Action<string> here instead of creating your own delegate.